application/models/Permission.php

```diff
@@ -567,7 +647,54 @@ public function hasSurveyPermission($iSurveyID, $sPermission, $sCRUD = 'read', $
         if (($sCRUD == 'delete' && $sPermission != 'survey')) { // Delete (token, reponse , question content …) need only allow update surveys
             $sGlobalCRUD = 'update';
         }
-        return $this->hasGlobalPermission('surveys', $sGlobalCRUD, $iUserID) || $this->hasPermission($iSurveyID, 'survey', $sPermission, $sCRUD, $iUserID);
+        return $this->hasGlobalPermission('surveys', $sGlobalCRUD, $iUserID)
+            || $this->hasSurveysInGroupPermission($oSurvey->gsid, 'surveys', $sGlobalCRUD, $iUserID)
+            || $this->hasPermission($iSurveyID, 'survey', $sPermission, $sCRUD, $iUserID);
+    }
+
+    /**
+     * Checks if a user has a certain permission in the given survey group
+     * 
+     * @param $iGroupId integer The survey ID
+     * @param $sPermission string Name of the permission
+     * @param $sCRUD string The permission detail you want to check on: 'create','read','update','delete','import' or 'export'
+     * @param $iUserID integer User ID - if not given the one of the current user is used
+     * @return bool True if user has the permission
+     */
+    public function hasSurveyGroupPermission($iGroupId, $sPermission, $sCRUD = 'read', $iUserID = null)
```

Comment:
Unit-test for this function would be nice. Let me look later.

Reply:
hasSurveyPermission before please ;) (or any Permission in fact : even GlobalSettings update)

Reply:
Yep yep.